### PR TITLE
Adding some basic functional tests

### PR DESCRIPTION
**Pull request by rynowak**
I ran into a blocking issue where page handlers aren't picked up, so I
fixed it. Logging another issue to follow up with better tests for this.

### CODE REVIEW

File: test/Microsoft.AspNetCore.Mvc.FunctionalTests/RazorPagesTest.cs [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+            // Act
+            var response = await Client.SendAsync(request);
+
+            // Assert
+            Assert.Equal(HttpStatusCode.OK, response.StatusCode);
+
+            var content = await response.Content.ReadAsStringAsync();
+            Assert.Equal("Hello, route!", content.Trim());
+        }
+        // HelloWorldWithPageModelHandler
```
Comment (ryanbrandenburg) [inline test/Microsoft.AspNetCore.Mvc.FunctionalTests/RazorPagesTest.cs]: ?

File: test/WebSites/RazorPagesWebSite/Startup.cs [hunk 2]
```diff
@@ -10,7 +10,7 @@ public class Startup
     {
         public void ConfigureServices(IServiceCollection services)
         {
-            services.AddMvc();
+            services.AddMvc().AddCookieTempDataProvider();
```
Comment (ryanbrandenburg) [inline test/WebSites/RazorPagesWebSite/Startup.cs]: What does this do?

File: test/Microsoft.AspNetCore.Mvc.FunctionalTests/RazorPagesTest.cs (hunk 1 above)
Comment (rynowak) [inline test/Microsoft.AspNetCore.Mvc.FunctionalTests/RazorPagesTest.cs]: Will delort this.

File: test/WebSites/RazorPagesWebSite/Startup.cs (hunk 2 above)
Comment (rynowak) [inline test/WebSites/RazorPagesWebSite/Startup.cs]: Will undo, and add back when I do the tempdata tests

File: src/Microsoft.AspNetCore.Mvc.RazorPages/Internal/PassThruRazorPageActivator.cs [hunk 3]
```diff
@@ -24,7 +24,11 @@ public void Activate(IRazorPage page, ViewContext context)
             {
                 var pageContext = (PageContext)context;
                 var vddType = typeof(ViewDataDictionary<>);
-                vddType = vddType.MakeGenericType(pageContext.ActionDescriptor.ModelTypeInfo.AsType());
+
+                var modelTypeInfo = pageContext.ActionDescriptor.ModelTypeInfo ?? pageContext.ActionDescriptor.PageTypeInfo;
+
```
Comment (pranavkm) [inline src/Microsoft.AspNetCore.Mvc.RazorPages/Internal/PassThruRazorPageActivator.cs]: Remove extra lines.

**Comment by rynowak** [comment]
19331f95326b18bd6f0400526074da934bf65cf0

**Pull request closed without merging**